STORM-675. Allow users to have storm-env.sh under config dir to set custom JAVA_HOME and other env variables.

harshach opened the pull request.

HeartSaVioR commented:
AFAIK, STORM-487 was intended to lay the foundation of unifying storm launch script.
Do we have plan to change storm.cmd to use storm.py?

harshach commented:
@HeartSaVioR yes but wanted to keep the .cmd files for at least a release or two as there might be users who are depend on these. Since storm.py is now windows compatabile we can phase out the current storm.cmd

HeartSaVioR commented:
Great! Looks good to me.
Btw, it's better to point out storm.py's actual changed lines since it's hard to find difference.

harshach commented:
@HeartSaVioR bin/storm which used to be a python script now moved to bin/storm.py and in place of bin/storm we added storm shell script to give users a way to configure storm env variables. Because of this renaming files you see the diff like that.

HeartSaVioR commented:
@harshach I know, I just want to point out that bin/storm is not only renamed but also modified a bit.

Parth-Brahmbhatt commented:
+1.

harshach commented:
@ptgoetz can you please take a look at this.

ptgoetz commented on bin/storm:
```diff
@@ -1,5 +1,7 @@
-
+#!/bin/bash
+#
+# Copyright 2014 The Apache Software Foundation
```
We shouldn't need the copyright year in the license header.

ptgoetz commented on conf/storm-env.sh:
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#
+# Copyright 2014 The Apache Software Foundation
```
Same as previous comment.

harshach commented:
@ptgoetz Thanks for the review. Removed the year from LICENSE.

ptgoetz commented:
+1. The diffs of `storm` vs. `storm.py` are minimal and related to windows compatibility.

For easy reference, here they are:

```
$ diff -w storm storm.py
37a38,40
> def is_windows():
>     return sys.platform.startswith('win')
>
174c177
<         JAVA_CMD, jvmtype, get_config_opts(),
---
>         "java", jvmtype, get_config_opts(),
184c187
<     else:
---
>     elif is_windows():
186a190,191
>     else:
>         os.execvp(JAVA_CMD, all_args)
```
